### PR TITLE
Generalize trusted publishing emails

### PR DESCRIPTION
**Pull request by woodruffw**
This makes the `trusted-publisher-added` and
`trusted-publisher-removed` email structures a little more
generic, allowing them to be re-used over both
GitHub and Google publishers. Future publishers will require
additional accommodations.

See https://github.com/pypi/warehouse/issues/13551.

Signed-off-by: William Woodruff <william@trailofbits.com>

### CODE REVIEW

File: tests/unit/accounts/test_views.py [hunk 1]
```diff
@@ -3627,7 +3627,7 @@ def test_add_pending_github_oidc_publisher(self, monkeypatch, db_request):
 
         assert db_request.session.flash.calls == [
             pretend.call(
-                "Registered a new publishing publisher to create "
+                "Registered a new pending publisher to create "
```
Comment (woodruffw) [inline tests/unit/accounts/test_views.py:3630]: NB: Noticed this small typo while working on this.

File: warehouse/oidc/models/google.py [hunk 2]
```diff
@@ -77,6 +77,13 @@ def __lookup_no_sub__(klass, signed_claims: SignedClaims) -> Query | None:
         __lookup_no_sub__,
     ]
 
+    @property
+    def publisher_name(self):
+        return "Google"
+
+    def publisher_url(self, claims=None):
+        return "https://accounts.google.com"
```
Comment (di) [inline warehouse/oidc/models/google.py:85]: For GitHub, this ends up being a link to a specific repository or SHA. I'm not sure this is useful for Google accounts. If these were always true Google accounts, we could link to `https://accounts.google.com/?authuser=user@gmail.com`, but I don't think that will work for service accounts.

**Review by di — APPROVED** [review]
LGTM aside from that one comment.

File: warehouse/oidc/models/google.py (hunk 2 above)
Comment (woodruffw) [reply warehouse/oidc/models/google.py:85]: Yeah, there may not be a great URL here. Perhaps we could link to Google's OIDC docs instead?

Comment (di) [reply warehouse/oidc/models/google.py:85]: I think we should really only provide this if it's specific to the identity used. I think it's probably fine for this to return `None` for now, and for the templates to handle this accordingly.

Comment (woodruffw) [reply warehouse/oidc/models/google.py:85]: Sounds good, will update.

Comment (woodruffw) [reply warehouse/oidc/models/google.py]: Okay, updated -- I updated a handful of the uses of `publisher_url()` to prepare them for providers where it'll be `None`, but some rendering uses will require a more in-depth look with the view/form changes (e.g. the trusted publisher table on each user/project).

**Comment by di** [comment]
@woodruffw You've got a failing test here now.

**Comment by woodruffw** [comment]
> @woodruffw You've got a failing test here now.

Thanks, fixed!

**Pull request merged**